### PR TITLE
Kamu 180 add ability to rename and to delete dataset

### PR DESCRIPTION
**Pull request by dmitriy-borzenko**

### CODE REVIEW

File: src/app/dataset-view/additional-components/settings-component/services/dataset-settings.service.ts [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+                    }
+                }
+            }),
+        );
+    }
+
+    public renameDataset(accountName: string, datasetId: string, newName: string): Observable<void> {
+        return this.datasetApi.renameDataset(datasetId, newName).pipe(
+            map((data: RenameDatasetMutation | undefined | null) => {
+                console.log("data", data);
```
Comment (EvilAlexei) [inline src/app/dataset-view/additional-components/settings-component/services/dataset-settings.service.ts:46]: Is this a leftover? Could you please remove it?

File: resources/schema.graphql [hunk 2]
```diff
@@ -1,392 +1,379 @@
 type AccessToken {
-  accessToken: String!
```
Comment (EvilAlexei) [inline resources/schema.graphql:2]: Change of the indents are necessary here?
I think it's better to use one type only tabs or only spaces.
As I see most of the files have spaces, could you use spaces here too?

File: src/app/dataset-view/additional-components/settings-component/settings.component.html [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+    <div class="Layout-sidebar">
+        <div class="pr-md-4 pr-0">
+            <div>
+                <nav aria-label="User settings" data-view-component="true" class="NavigationList">
+                    <ul
+                        role="list"
+                        aria-label="User settings"
+                        nav_classes="org-sub-menu"
+                        data-view-component="true"
+                        class="ActionList"
```
Comment (EvilAlexei) [inline src/app/dataset-view/additional-components/settings-component/settings.component.html:11]: I strongly recommend not to use cameCase for class names,
please rename to `action-list`

File: src/app/dataset-view/additional-components/settings-component/settings.component.html [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+                    >
+                        <li
+                            data-item-id=""
+                            data-view-component="true"
+                            [ngClass]="{
+                                'ActionList-item--navActive': true
```
Comment (EvilAlexei) [inline src/app/dataset-view/additional-components/settings-component/settings.component.html:17]: Same here, please don't use camelCase in class names.
```suggestion
                                'action-list-item--nav-active': true
```

Same for the rest of the page, and styles

File: src/app/dataset-view/additional-components/settings-component/settings.component.html (hunk 3 above)
Comment (EvilAlexei) [inline src/app/dataset-view/additional-components/settings-component/settings.component.html:11]: I strongly recommend not to use camelCase for class names,
please rename to action-list

File: src/app/dataset-view/additional-components/settings-component/settings.component.html (hunk 4 above)
Comment (EvilAlexei) [inline src/app/dataset-view/additional-components/settings-component/settings.component.html:17]: Same here, please don't use camelCase in class names.
```suggestion
                                'action-list-item--nav-active': true
```

Same for the rest of the page, and styles

File: resources/schema.graphql (hunk 2 above)
Comment (zaychenko-sergei) [reply resources/schema.graphql:2]: This is file that should not be manually touched in general, it's created by scripts.
Looks like the file was explicitly saved in the editor.

File: src/app/dataset-view/additional-components/settings-component/services/dataset-settings.service.ts [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+                            datasetName: newName,
+                        })
+                        .subscribe();
+                    this.navigationService.navigateToDatasetView({
+                        accountName,
+                        datasetName: newName,
+                        tab: DatasetViewTypeEnum.Overview,
+                    });
+                } else {
+                    if (data) {
```
Comment (zaychenko-sergei) [inline src/app/dataset-view/additional-components/settings-component/services/dataset-settings.service.ts:60]: Add better handling for naming collision errors.

File: src/app/dataset-view/additional-components/settings-component/settings.component.html (hunk 3 above)
Comment (zaychenko-sergei) [reply src/app/dataset-view/additional-components/settings-component/settings.component.html:11]: Maybe also check if there are other cases like this in the old code

**Review by zaychenko-sergei — CHANGES_REQUESTED** [review]
Don't forget solid tests

File: src/app/dataset-view/dataset-view.component.sass [hunk 6]
```diff
@@ -4,7 +4,7 @@
 @import 'node_modules/prismjs/plugins/line-numbers/prism-line-numbers'
 @import 'node_modules/prismjs/plugins/line-highlight/prism-line-highlight'
 
-mat-button-toggle-group
+#mat-button-toggle-group
```
[the remark this answers is not in the document]
Comment (dmitriy-borzenko) [reply src/app/dataset-view/dataset-view.component.sass:7]: Fixed, used the class

File: src/app/dataset-view/additional-components/settings-component/settings.component.html (hunk 4 above)
Comment (dmitriy-borzenko) [reply src/app/dataset-view/additional-components/settings-component/settings.component.html:17]: Fixed

File: src/app/dataset-view/additional-components/settings-component/services/dataset-settings.service.spec.ts [hunk 7]
```diff
@@ -0,0 +1,89 @@
...
+        navigationService = TestBed.inject(NavigationService);
+        datasetApi = TestBed.inject(DatasetApi);
+        modalService = TestBed.inject(ModalService);
+    });
+
+    it("should be created", () => {
+        expect(service).toBeTruthy();
+    });
+
+    it("should check detete dataset with succes", () => {
```
Comment (zaychenko-sergei) [inline src/app/dataset-view/additional-components/settings-component/services/dataset-settings.service.spec.ts:41]: suces**s**

File: src/app/dataset-view/additional-components/settings-component/services/dataset-settings.service.ts [hunk 8]
```diff
@@ -0,0 +1,90 @@
...
+                        this.proccesRenameError(data);
+                    }
+                }
+            }),
+        );
+    }
+
+    private proccesRenameError(data: RenameDatasetMutation): void {
+        if (data.datasets.byId?.rename.__typename === "RenameResultNameCollision") {
```
Comment (zaychenko-sergei) [inline src/app/dataset-view/additional-components/settings-component/services/dataset-settings.service.ts:78]: Handle No Changes in similar way

File: src/app/dataset-view/additional-components/settings-component/services/dataset-settings.service.spec.ts (hunk 7 above)
Comment (dmitriy-borzenko) [reply src/app/dataset-view/additional-components/settings-component/services/dataset-settings.service.spec.ts:41]: Fixed

**Review by zaychenko-sergei — APPROVED** [review]

File: src/app/dataset-view/additional-components/settings-component/settings.component.sass [hunk 9]
```diff
@@ -0,0 +1,139 @@
...
+        top: calc(50% - 12px)
+        left: -8px
+        width: 4px
+        height: 24px
+        content: ""
+        background: $app-color-accent-fg
+        border-radius: 6px
+
+.action-list-item--navActive a
+    font-weight: 600
```
Comment (EvilAlexei) [inline src/app/dataset-view/additional-components/settings-component/settings.component.sass:79]: 1. please rename `navActive` - `nav-active`
2. I think it's better to replace this to the `.action-list-item` section above
```
.action-list-item

  &--nav-active
  ...
  
    &:after
    ...
    
    a
    ...
```

File: src/app/dataset-view/additional-components/settings-component/settings.component.sass [hunk 10]
```diff
@@ -0,0 +1,139 @@
...
+    background: $app-color-action-list-item-inline-divider
+    border: 0
+
+.action-list-sectionDivider:not(:empty)
+    display: flex
+    padding: 6px 8px
+    font-size: 12px
+    font-weight: 500
+    color: $app-color-fg-muted
+    flex-direction: column
```
Comment (EvilAlexei) [inline src/app/dataset-view/additional-components/settings-component/settings.component.sass:96]: Same as above, I suggest to use sass nesting, and rename `sectionDivider` -> `section-divider`

```
.action-list-section-divider

  &:empty
  ...
  
  &:not(:empty)
  ...
```

**Review by EvilAlexei — CHANGES_REQUESTED** [review]

**Pull request merged**